Modules/_struct.c

```diff
@@ -1930,15 +1930,40 @@ s_pack_into(PyObject *self, PyObject **args, Py_ssize_t nargs, PyObject *kwnames
         return NULL;
     }
 
+    /* Check that negative offset is low enough to fit data */
+    if (offset < 0 && offset + soself->s_size > 0) {
```

Comment:
I think three following `if`'s can be merged into one `if (offset < 0)`. This will speed up common case of non-negative `offset`.